### PR TITLE
Added provider support for deleting all empty cards for a given model

### PR DESCRIPTION
**Pull request by marcardar**

### CODE REVIEW

**Comment by marcardar** [comment]
I still wonder about having a `/models/*/cards` pattern because that would work well in future queries or, for example, counting the number of cards for a model.

File: AnkiDroid/src/main/java/com/ichi2/anki/provider/CardContentProvider.java [hunk 1]
```diff
@@ -169,6 +171,8 @@ public String getType(Uri uri) {
                 return FlashCardsContract.Model.CONTENT_TYPE;
             case MODELS_ID:
                 return FlashCardsContract.Model.CONTENT_ITEM_TYPE;
+            case MODELS_ID_EMPTY_CARDS:
+                return FlashCardsContract.Model.CONTENT_TYPE;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/provider/CardContentProvider.java]: This should really be
`FlashCardsContract.Card.CONTENT_ITEM_TYPE` I guess

Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/provider/CardContentProvider.java]: I mean `FlashCardsContract.Card.CONTENT_TYPE`

Comment (marcardar) [inline AnkiDroid/src/main/java/com/ichi2/anki/provider/CardContentProvider.java]: Yep, well spotted. Fixed.

**Comment by timrae** [comment]
> I still wonder about having a /models/*/cards pattern because that would work well in future queries or, for example, counting the number of cards for a model.

Let's cross that bridge when we come to it. If we do decided that it's necessary then it will be separate from the EMPTY_CARDS URI so no need to discuss it here.

**Pull request merged**